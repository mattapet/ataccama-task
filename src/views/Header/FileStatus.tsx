//#region imports
import React from 'react';
import styled from 'styled-components';
import { Button } from 'antd';
//#endregion

//#region Component interfaces
export interface Props {
  /**
   * Currently selected file. If no file is selected, the value is supposed
   * to be `null`.
   */
  readonly file: File | null;
  /**
   * Callback to clear the currently selected file.
   */
  readonly onClear: () => any;
}
//#endregion

//#region Styled
const Wrapper = styled.div`
  color: whitesmoke;
  * {
    color: whitesmoke;
    margin-right: 0.25em;
    margin-left: 0.25em;
  }
`;
//#endregion

/**
 * Displays currently selected file with an option to clear the selection.
 */
const FileStatus: React.FunctionComponent<Props> = props => (
  <Wrapper>
    <span>{props.file ? props.file.name : <em>No File Selected</em>}</span>
    <Button type="ghost" shape="round" onClick={props.onClear}>
      Clear
    </Button>
  </Wrapper>
);

export default FileStatus;
