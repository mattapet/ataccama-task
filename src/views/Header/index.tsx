//#region imports
import React, { useContext } from 'react';
import styled from 'styled-components';
import { Layout } from 'antd';
import { FileContext } from 'stores/FileStore';

import FileStatus from './FileStatus';
//#endregion

//#region Styled
const LayoutHeader = styled(Layout.Header)`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`;

const Title = styled.h1`
  color: whitesmoke;
  font-family: 'Courier New', Courier, monospace;
`;
//#endregion

/**
 * Global application header.
 *
 * Component displays title of the application as well as file status based
 * on `FileContext`.
 */
const Header: React.FunctionComponent = () => {
  const { file, onClear } = useContext(FileContext);
  return (
    <LayoutHeader>
      <Title>ataccama-task</Title>
      <FileStatus file={file} onClear={onClear} />
    </LayoutHeader>
  );
};

export default Header;
