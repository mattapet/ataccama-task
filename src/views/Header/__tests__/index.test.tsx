//#region imports
import React from 'react';
import renderer from 'react-test-renderer';

import Header from '../index';
//#endregion

let container: HTMLDivElement | null = null;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  container && document.body.removeChild(container);
  container = null;
});

it('renders correctly', () => {
  // Arrange
  // Act
  const component = renderer.create(<Header />);
  // Assert
  expect(component).toMatchSnapshot();
});
