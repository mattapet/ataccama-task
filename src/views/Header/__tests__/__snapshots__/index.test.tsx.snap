// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="sc-bwzfXH htoLMg ant-layout-header"
>
  <h1
    className="sc-htpNat gQPDbF"
  >
    ataccama-task
  </h1>
  <div
    className="sc-bdVaJa NIJn"
  >
    <span>
      <em>
        No File Selected
      </em>
    </span>
    <button
      className="ant-btn ant-btn-ghost ant-btn-round"
      onClick={[Function]}
      type="button"
    >
      <span>
        Clear
      </span>
    </button>
  </div>
</header>
`;
